hastory-data/src/Data/Hastory/Types.hs

```diff
@@ -44,6 +48,16 @@ instance Hashable Entry
 
 instance Validity Entry
 
+instance GenValid Entry where
+  genValid = Entry
+               <$> genValid -- Gen Text
+               <*> genValid -- Gen (Path Abs Dir)
+               <*> genValid -- Gen UTCTime
+               <*> genValid -- Gen Text
+               <*> genValid -- Gen Text
+
+  shrinkValid entry = [entry] -- No Shrinking
```

Comment:
The shrinking function must never return itself.
In this case you can just use `shrinkValidStructurally`.